lib/util/query.js

```diff
@@ -148,22 +149,52 @@ export function getDefaultQuery (config) {
   return defaultQuery
 }
 
+/**
+ * Geocode utility for returning the first result for the provided place name text.
+ * @param  {String} text - text to search
+ * @param  {Object} geocoderConfig
+ * @return {Location}
+ */
+async function getFirstGeocodeResult (text, geocoderConfig) {
```

Comment:
Is the underlying geocoder API async?

Reply:
The geocoder API makes asynchronous fetches, so if we want to be able to use the result inline (instead of in a then clause), we need to make this function async.